src/bytes_ser_de.rs

```diff
@@ -34,19 +37,22 @@ pub trait Serializable: Sized {
     #[inline]
     fn try_from_bytes(bytes: &[u8]) -> Result<Self, Self::Error> {
         if bytes.is_empty() {
-            return Err(
-                CryptoCoreError::InvalidSize("Given byte string is empty".to_string()).into(),
-            );
+            return Err(CryptoCoreError::DeserialisationSizeError {
+                given: 0,
+                expected: size_of::<Self>(),
```

Comment:
No, this size is wrong because we may write LEB128 sizes in addition of the object's data. Because of the LEB128 format, you can only know a lower bound on the expected size (LEB128 sizes can range from 1 to several bytes).

Reply:
Do we encode with LEB128 fixed sized arrays?